src/game_api/script/lua_backend.cpp

```diff
@@ -713,6 +716,156 @@ void LuaBackend::post_entity_spawn(Entity* entity, int spawn_type_flags)
         }
     }
 }
+void LuaBackend::pre_render_hud()
+{
+    if (!get_enabled())
+        return;
+
+    /// Runs before the HUD is drawn on screen. You can draw text and textures on screen by using the provided render_ctx parameter
+    sol::optional<sol::function> on_render_pre_hud = lua["on_render_pre_hud"];
+
+    VanillaRenderContext render_ctx;
+
+    if (on_render_pre_hud)
+    {
+        on_render_pre_hud.value()(render_ctx);
+    }
+
+    for (auto& [id, callback] : callbacks)
+    {
+        auto now = get_frame_count();
+        if (callback.screen == ON::RENDER_PRE_HUD)
+        {
+            handle_function(callback.func, render_ctx);
+            callback.lastRan = now;
+        }
+    }
+}
+void LuaBackend::post_render_hud()
+{
+    if (!get_enabled())
+        return;
+
+    /// Runs after the HUD is drawn on screen. You can draw text and textures on screen by using the provided render_ctx parameter
+    sol::optional<sol::function> on_render_post_hud = lua["on_render_post_hud"];
+
+    VanillaRenderContext render_ctx;
+
+    if (on_render_post_hud)
+    {
+        on_render_post_hud.value()(render_ctx);
+    }
+
+    for (auto& [id, callback] : callbacks)
+    {
+        auto now = get_frame_count();
+        if (callback.screen == ON::RENDER_POST_HUD)
+        {
+            handle_function(callback.func, render_ctx);
+            callback.lastRan = now;
+        }
+    }
+}
+void LuaBackend::pre_render_pause_menu()
+{
+    if (!get_enabled())
+        return;
+
+    /// Runs before the pause menu is drawn on screen. You can draw text and textures on screen by using the provided render_ctx parameter
+    sol::optional<sol::function> on_render_pre_pause_menu = lua["on_render_pre_pause_menu"];
```

Comment:
Are we sure we want to add more of those kinds of callbacks? We said in the past that they are a bit too magic so we stick to `set_callback`. It would also make the code a tiny bit less complex.